src/config/config.hpp

```
@@ -0,0 +1,88 @@
+ * Types
+ **********************************************************************************************************************/
+
+/*
+ * Identifier identifier plugin parameters.
+ */
+struct Identifier {
+    std::string        mPlugin;
+    Poco::Dynamic::Var mParams;
+};
+
+/*
+ * RemoteIAM remote IAM parameters.
+ */
+struct RemoteIAM {
+    std::string mNodeID;
+    std::string mURL;
+    Duration    mRequestTimeout;
+};
+
+/*
+ * ModuleConfig module configuration.
+ */
+struct ModuleConfig {
+    std::string              mID;
+    std::string              mPlugin;
+    std::string              mAlgorithm;
+    int                      mMaxItems;
+    std::vector<std::string> mExtendedKeyUsage;
+    std::vector<std::string> mAlternativeNames;
+    bool                     mDisabled;
+    bool                     mSkipValidation;
+    Poco::Dynamic::Var       mParams;
+};
+
+/*
+ * Config instance.
+ */
+struct Config {
+    std::string               mIAMPublicServerURL;
+    std::string               mIAMProtectedServerURL;
+    std::string               mNodeID;
+    std::string               mNodeType;
+    std::string               mCACert;
+    std::string               mCertStorage;
+    std::string               mWorkingDir;
+    std::vector<ModuleConfig> mCertModules;
+    std::vector<std::string>  mFinishProvisioningCmdArgs;
+    std::vector<std::string>  mDiskEncryptionCmdArgs;
+    bool                      mEnablePermissionsHandler;
+    Identifier                mIdentifier;
+    std::vector<RemoteIAM>    mRemoteIAMs;
+};
+
+/*******************************************************************************
+ * Functions
+ ******************************************************************************/
+
+/*
+ * Parse config from file.
+ *
+ * @param filename config file name.
+ * @return config instance.
+ */
+aos::RetWithError<Config> ParseConfig(const std::string& filename);
```

Throw aos exception instead of return error.

---

Decide to return errors in external functions or methods